syntax = "proto3";

service Booking {
    rpc GetJson(Empty) returns (BookingList) {}
    rpc GetBookings(UserID) returns (BookingList) {}
    rpc AddBooking(AddBookingRequest) returns (Empty) {}
}

message UserID {
    string id = 1;
}

message BookingList {
    repeated _Booking bookings = 1;
}

message AddBookingRequest {
    string userid = 1;
    string date = 2;
    MovieList movies_id = 3;
}


message _Booking {
    string date = 1;
    MovieList movies_id = 2;
}

message MovieList {
    repeated string movies_id = 1;
}

message Empty {
}